core/src/main/scala/kafka/security/auth/SimpleAclAuthorizer.scala

```diff
@@ -246,8 +246,11 @@ class SimpleAclAuthorizer extends Authorizer with Logging {
   }
 
   private def logAuditMessage(principal: KafkaPrincipal, authorized: Boolean, operation: Operation, resource: Resource, host: String) {
-    val permissionType = if (authorized) "Allowed" else "Denied"
-    authorizerLogger.debug(s"Principal = $principal is $permissionType Operation = $operation from host = $host on resource = $resource")
+    val logStr = s"Principal = $principal is %s Operation = $operation from host = $host on resource = $resource"
+    if (authorized)
+      authorizerLogger.debug(logStr.format("Allowed"))
+    else
+      authorizerLogger.info(logStr.format("Denied"))
```

Comment:
This is odd. The `if/else` should be to define a val and that can then be used in the interpolated string itself.

Reply:
Actually, we should if checking the log level before computing the string in case it's not needed.

Reply:
@ijuma Yes, we can get some performance gain by checking log-level before computing the string. Updated the code. Pl check